lightning/src/ln/channelmanager.rs

```diff
@@ -3797,6 +3803,15 @@ impl<'a, ChanSigner: ChannelKeys + Readable, M: Deref, T: Deref, K: Deref, F: De
 			per_peer_state.insert(peer_pubkey, Mutex::new(peer_state));
 		}
 
+        let event_count: u64 = Readable::read(reader)?;
+        let mut pending_events: Vec<events::Event> = Vec::with_capacity(event_count as usize);
```

Comment:
This will OOM on an invalid input. You may want something like:
```suggestion
        let mut pending_events: Vec<events::Event> = Vec::with_capacity(cmp::min(event_count as usize, mem::size_of::<events::Event>()));
```